clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -102,6 +104,11 @@ public int validBytes() {
      * @return A FilterResult with a summary of the output (for metrics)
      */
     public FilterResult filterTo(LogEntryFilter filter, ByteBuffer buffer) {
+        return filterTo(shallowEntries(), filter, buffer);
+    }
+
+    private static FilterResult filterTo(Iterable<ByteBufferLogEntry> fromShallowEntries, LogEntryFilter filter,
+                                       ByteBuffer destinationBuffer) {
```

Comment:
nit: I'd name the parameters consistently, so the other `filterTo` should also use `destinationBuffer`.